crates/shared/src/http_solver/model.rs

```diff
@@ -338,6 +338,15 @@ pub struct ExecutionPlanCoordinatesModel {
     pub position: u32,
 }
 
+/// The result of a submission process for a winning solver
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SubmissionResult {
+    Success(H256),
+    Revert(H256),
+    Fail,
```

Comment:
Fail -> No transaction was mined for this auction.
It could happen for various reasons but I think more granularity is not needed for external solvers.